File: src/Counsel.FantasyClient/Models/AdvancedResults.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Counsel.FantasyClient.Models
{
    public class AdvancedResults
    {
        [JsonProperty("QB")]
        public List<AdvancedPlayerResult> Quarterbacks { get; set; }

        [JsonProperty("RB")]
        public List<AdvancedPlayerResult> RunningBacks { get; set; }

        [JsonProperty("WR")]
        public List<AdvancedPlayerResult> WideReceivers { get; set; }

        [JsonProperty("TE")]
        public List<AdvancedPlayerResult> TightEnds { get; set; }

        [JsonProperty("K")]
        public List<AdvancedPlayerResult> Kickers { get; set; }

        [JsonProperty("DEF")]
        public List<AdvancedPlayerResult> Defenses { get; set; }
    }
}

File: src/Counsel/FantasyClient/AdvancedResults.swift
//
//  AdvancedResults.swift
//  Counsel
//
//  Created by Han Hossain on 6/8/19.
//  Copyright © 2019 Han Hossain. All rights reserved.
//

import Foundation

struct AdvancedResults {
	let quarterbacks: [AdvancedPlayerResult]?
	let runningBacks: [AdvancedPlayerResult]?
	let wideReceivers: [AdvancedPlayerResult]?
	let tightEnds: [AdvancedPlayerResult]?
	let kickers: [AdvancedPlayerResult]?
	let defenses: [AdvancedPlayerResult]?
}

extension AdvancedResults {
	init(json: [String : Any]) throws {
		let quarterbacksJson = json["QB"] as? [[String : Any]]
		quarterbacks = try quarterbacksJson?.map { try AdvancedPlayerResult(json: $0) }

		let runningBacksJson = json["RB"] as? [[String : Any]]
		runningBacks = try runningBacksJson?.map { try AdvancedPlayerResult(json: $0) }

		let wideReceiversJson = json["WR"] as? [[String : Any]]
		wideReceivers = try wideReceiversJson?.map { try AdvancedPlayerResult(json: $0) }

		let tightEndsJson = json["TE"] as? [[String : Any]]
		tightEnds = try tightEndsJson?.map { try AdvancedPlayerResult(json: $0) }

		let kickersJson = json["K"] as? [[String : Any]]
		kickers = try kickersJson?.map { try AdvancedPlayerResult(json: $0) }

		let defensesJson = json["DEF"] as? [[String : Any]]
		defenses = try defensesJson?.map { try AdvancedPlayerResult(json: $0) }
	}
}
